debezium-server-voyager/debezium-server-voyagerexporter/src/main/java/io/debezium/server/ybexporter/EventCache.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.server.ybexporter;
+
+import java.io.IOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EventCache {
+    private static final Logger LOGGER = LoggerFactory.getLogger(EventQueue.class);
+    private static final String EOF_MARKER = "\\.";
+    String dataDir;
+    private LinkedList<String> eventQueue = new LinkedList<>();
+    private Set<Integer> hashCache = new HashSet<>();
+    private long currentQueueSegmentIndex = 0;
+    private static final String QUEUE_SEGMENT_FILE_NAME = "segment";
+    private static final String QUEUE_SEGMENT_FILE_EXTENSION = "ndjson";
+    private static final String QUEUE_FILE_DIR = "queue";
+    private long maxCacheSize = 10000;
+    private QueueSegment currentQueueSegment;
+
+    public EventCache(String dataDir) {
+        this.dataDir = dataDir;
+        recoverCacheFromDisk();
+    }
+
+    public void recoverCacheFromDisk() {
+        hashCache = new HashSet<>();
+        recoverLatestQueueSegment();
+        if (currentQueueSegment == null) {
+            LOGGER.info("No queue segment found. Nothing to recover into event cache");
+            return;
+        }
+        while (true) {
+            String events = currentQueueSegment.readAllEvents();
+            if (events == null) {
+                // Segment is empty
+                break;
+            }
+            String[] eventArray = events.split("\n");
+            for (int i = eventArray.length - 1; i >= 0; i--) {
+                if (eventArray[i].equals(EOF_MARKER) || eventArray[i].isEmpty() || eventArray[i].isBlank()) {
+                    continue;
+                }
+
+                String cacheMetadata = getCacheMetadata(eventArray[i]);
```

Comment:
Parse the current event using any ndjson parser. And then extract "ID" of the message.

Reply:
Now the value from `event_id` is extracted from the segment file and cached:
`"event_id":"0100100073930000,1,1"`